RecoHGCal/TICL/plugins/PatternRecognitionbyCLUE3D.cc

```diff
@@ -645,20 +769,29 @@ int PatternRecognitionbyCLUE3D<TILES>::findAndAssignTracksters(
   unsigned int nTracksters = 0;
 
   std::vector<std::pair<int, int>> localStack;
+  auto critical_transverse_distance = useAbsoluteProjectiveScale_ ? criticalXYDistance_ : criticalEtaPhiDistance_;
   // find cluster seeds and outlier
   for (unsigned int layer = 0; layer < 2 * rhtools_.lastLayer(); layer++) {
     auto &clustersOnLayer = clusters_[layer];
     unsigned int numberOfClusters = clustersOnLayer.x.size();
     for (unsigned int i = 0; i < numberOfClusters; i++) {
       // initialize clusterIndex
       clustersOnLayer.clusterIndex[i] = -1;
-      bool isSeed =
-          (clustersOnLayer.delta[i] > criticalEtaPhiDistance_) && (clustersOnLayer.rho[i] >= criticalDensity_);
-      bool isOutlier = (clustersOnLayer.delta[i] > outlierMultiplier_ * criticalEtaPhiDistance_) &&
+      bool isSeed = (clustersOnLayer.delta[i].first > critical_transverse_distance ||
+                     clustersOnLayer.delta[i].second > criticalZDistanceLyr_) &&
+                    (clustersOnLayer.rho[i] >= criticalDensity_) &&
+                    (clustersOnLayer.energy[i] / clustersOnLayer.rho[i] > criticalSelfDensity_);
+      if (!clustersOnLayer.isSilicon[i]) {
+        isSeed = (clustersOnLayer.delta[i].first > clustersOnLayer.radius[i] ||
+                  clustersOnLayer.delta[i].second > criticalZDistanceLyr_) &&
+                 (clustersOnLayer.rho[i] >= criticalDensity_) &&
+                 (clustersOnLayer.energy[i] / clustersOnLayer.rho[i] > criticalSelfDensity_);
+      }
```

Comment:
```suggestion
      bool isSeed = clustersOnLayer.isSilicon[i]
                ? (clustersOnLayer.delta[i].first > critical_transverse_distance ||
                    clustersOnLayer.delta[i].second > criticalZDistanceLyr_) &&
                    (clustersOnLayer.rho[i] >= criticalDensity_) &&
                     (clustersOnLayer.energy[i] / clustersOnLayer.rho[i] > criticalSelfDensity_)
                : (clustersOnLayer.delta[i].first > clustersOnLayer.radius[i] ||
                   clustersOnLayer.delta[i].second > criticalZDistanceLyr_) &&
                   (clustersOnLayer.rho[i] >= criticalDensity_) &&
                    (clustersOnLayer.energy[i] / clustersOnLayer.rho[i] > criticalSelfDensity_);
```

Reply:
The ratio of Silicon channels to Scintillator is 6M to 400K. Not sure this will make any difference.